src/sidebar/store/modules/annotations.js

```diff
@@ -137,14 +152,58 @@ const update = {
     };
   },
 
+  CLEAR_ANNOTATIONS: function () {
+    return { annotations: [] };
+  },
+
+  FOCUS_ANNOTATIONS: function (state, action) {
+    return { focused: toTrueMap(action.focusedTags) };
+  },
+
+  HIDE_ANNOTATION: function (state, action) {
+    const anns = state.annotations.map(function (ann) {
+      if (ann.id !== action.id) {
+        return ann;
+      }
+      return Object.assign({}, ann, { hidden: true });
```

Comment:
```suggestion
      return { ...ann, hidden: true }
```

This is just a suggestion - I know you're just moving existing code here.